hazelcast/serialization/service.py

```diff
@@ -496,6 +501,22 @@ def register_from_super_type(self, obj_type, super_type) -> typing.Optional[Stre
             self.safe_register_serializer(serializer, obj_type)
         return serializer
 
+    def validate(self):
+        """
+        Makes sure that the classes registered as Compact serializable are not
+        overriding the default serializers.
+
+        Must be called in the constructor of the serialization service after it
+        completes registering default serializers.
+        """
+        for compact_type in self._compact_types:
+            if compact_type in self._constant_type_dict:
+                raise HazelcastSerializationError(
```

Comment:
Java client throws `IllegalArgumentException` here.

Reply:
That makes more sense, let me change it 👍 BTW, do you think using `ValueError` in Python for this case, instead of `IllegalArgumentError` (which is one of our custom errors) would be more idiomatic?

Reply:
I'm leaning towards `IllegalArgumentError` to make it more similar to the reference implementation. In the Python docs there's this at: https://docs.python.org/3/library/exceptions.html
```
Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).
```
So, `ValueError` seems to be more appropriate when there's a problem with the value itself.

Reply:
https://github.com/hazelcast/hazelcast-python-client/pull/610/commits/b4d0105c9b4f075e904e92841fd3cc8ca39526f6